cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/KryoSerializationTests.java

```diff
@@ -371,25 +374,65 @@ public void testSslConfig()
     @Test
     public void testStorageTransportConfiguration()
     {
-        final StorageTransportConfiguration config = new StorageTransportConfiguration(
+        StorageTransportConfiguration config = new StorageTransportConfiguration(
         "writeBucket",
         "writeRegion",
         "readBucket",
         "readRegion",
         "prefix",
-        new StorageCredentialPair(
-        new StorageCredentials("keyId1", "secret1", "sessionToken1"),
-        new StorageCredentials("keyId2", "secret2", "sessionToken2")
+        new StorageCredentialPair("writeRegion",
+                                  new StorageCredentials("keyId1", "secret1", "sessionToken1"),
+                                  "readRegion",
+                                  new StorageCredentials("keyId2", "secret2", "sessionToken2")
         ),
-        ImmutableMap.of("tag1", "tagVal1", "tag2", "tagVal2")
-        );
+        ImmutableMap.of("tag1", "tagVal1", "tag2", "tagVal2"));
 
-        StorageTransportConfiguration deserialized;
-        try (Output out = serialize(config))
+        testSerDeserStorageTransportConfiguration(config);
+    }
+
+    @Test
+    public void testStorageTransportConfigurationWithMultiClusters()
```

Comment:
I think they are placed all in one class because of the below code block for convenience. 

I agree that we can separate the test cases into their own test classes, for many benefits.

```java
    static
    {
        new KryoRegister().registerClasses(KRYO);
        new SbwKryoRegistrator().registerClasses(KRYO);
    }
```